am/src/commands/instrument.rs

```diff
@@ -0,0 +1,115 @@
+use am_list::Language;
+use anyhow::Context;
+use clap::{Args, Subcommand};
+use std::path::PathBuf;
+use tracing::info;
+
+#[derive(Args)]
+pub struct Arguments {
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand)]
+enum Command {
+    /// Instrument functions in a single project, giving the language implementation
+    ///
+    /// IMPORTANT: This will add code in your files! If you want to easily
+    /// undo the effects of this command, stage your work in progress (using `git add` or similar)
+    /// So that a command like `git restore .` can undo all unstaged changes, leaving your work
+    /// in progress alone.
+    Single(SingleProject),
+    /// Instrument functions in all projects under the given directory, detecting languages on a best-effort basis.
+    ///
+    /// IMPORTANT: This will add code in your files! If you want to easily
+    /// undo the effects of this command, stage your work in progress (using `git add` or similar)
+    /// So that a command like `git restore .` can undo all unstaged changes, leaving your work
+    /// in progress alone.
+    All(AllProjects),
+}
+
+#[derive(Args)]
+struct SingleProject {
+    /// Language to detect autometrics functions for. Valid values are:
+    /// - 'rust' or 'rs' for Rust,
+    /// - 'go' for Golang,
+    /// - 'typescript', 'ts', 'javascript', or 'js' for Typescript/Javascript,
+    /// - 'python' or 'py' for Python.
+    #[arg(short, long, value_name = "LANGUAGE", verbatim_doc_comment)]
+    language: Language,
+    /// Root of the project to start the search on:
+    /// - For Rust projects it must be where the Cargo.toml lie,
```

Comment:
what does it do for workspaces? they do end up with multiple `Cargo.toml` then

Reply:
For workspaces, we assume that the project is where the workspace `Cargo.toml` file is (where there is the `[members]` array live). For "single" projects it shouldn't be an issue anyway because we give the root ourselves (so if you want to only work with a subcrate you can pass the path to the subcrate). It starts becoming an issue with the "all" project and the automatic root detection, I probably need to add tests eventually to make the "all" version more robust and predictable